python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -751,9 +751,9 @@ def key(node_and_num_running_replicas_of_all_deployments):
         ):
             if node_id not in node_to_running_replicas_of_target_deployment:
                 continue
-            for running_replica in node_to_running_replicas_of_target_deployment[
-                node_id
-            ]:
+            for running_replica in reversed(
+                list(node_to_running_replicas_of_target_deployment[node_id])
+            ):
```

Comment:
since `node_to_running_replicas_of_target_deployment[node_id]` is a set, we dont get any guarantees that its going to stop replicas in reverse order. This needs a different implementation

Reply:
i dont think `node_id` is chronologically increasing, but I could be wrong, can you look into that. If I am right then the sorted node_id will not help with the task you set out to achieve.